From d96975abed5530beb293000ee043ea36f3efed46 Mon Sep 17 00:00:00 2001
From: Lei Wen <leiwen@marvell.com>
Date: Sun, 1 Sep 2013 17:44:00 -0500
Subject: [PATCH 176/338] sched: change cpu_avg_load_per_task using
 h_nr_running

Date	Sun, 18 Aug 2013 16:25:16 +0800

Since cpu_avg_load_per_task is used only by cfs scheduler, its meaning
should present the average cfs type task load in the current run queue.
Thus we change it to h_nr_running for well presenting its meaning.

Signed-off-by: Lei Wen <leiwen@marvell.com>
Signed-off-by: Paul Reioux <reioux@gmail.com>
---
 kernel/sched/fair.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
index 0f46f63..1d7afec 100644
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -2329,7 +2329,7 @@ static unsigned long power_of(int cpu)
 static unsigned long cpu_avg_load_per_task(int cpu)
 {
 	struct rq *rq = cpu_rq(cpu);
-	unsigned long nr_running = ACCESS_ONCE(rq->nr_running);
+	unsigned long nr_running = ACCESS_ONCE(rq->cfs.h_nr_running);
 
 	if (nr_running)
 		return rq->load.weight / nr_running;
-- 
1.8.1.2

